pkg/ebpftracer/tracer_decode.go

```diff
@@ -97,18 +55,113 @@ func (t *Tracer) decodeAndExportEvent(ctx context.Context, ebpfMsgDecoder *decod
 		return fmt.Errorf("cannot get container for cgroup %d: %w", eventCtx.CgroupID, err)
 	}
 
+	rawEventTime := eventCtx.Ts
 	eventCtx.Ts = t.bootTime + eventCtx.Ts
 	event := &types.Event{
 		Context:   &eventCtx,
 		Container: container,
 		Args:      parsedArgs,
 	}
 
+	switch eventId {
+	case events.SchedProcessExec:
+		if eventCtx.Pid == 1 {
+			t.cfg.MountNamespacePIDStore.ForceAddToBucket(proc.NamespaceID(eventCtx.MntID), eventCtx.NodeHostPid)
+		} else {
+			t.cfg.MountNamespacePIDStore.AddToBucket(proc.NamespaceID(eventCtx.MntID), eventCtx.NodeHostPid)
+		}
+
+		parentStartTime := time.Duration(0)
+		if eventCtx.Ppid != 0 {
+			// We only set the parent start time, if we know the parent PID comes from the same NS.
+			parentStartTime = time.Duration(eventCtx.ParentStartTime) * time.Nanosecond
+		}
+		execArgs, ok := parsedArgs.(types.SchedProcessExecArgs)
+		if !ok {
+			t.log.Errorf("expected types.SchedProcessExecArgs, but got: %t", parsedArgs)
+			return nil
+		}
+		processStartTime := time.Duration(eventCtx.StartTime) * time.Nanosecond
+
+		t.cfg.ProcessTreeCollector.ProcessStarted(
+			system.GetBootTime().Add(time.Duration(rawEventTime)),
+			container.ID,
+			processtree.Process{
+				PID:             proc.PID(eventCtx.Pid),
+				StartTime:       processStartTime.Truncate(time.Second),
```

Comment:
Why StartTime and ParentStartTime is trucated to second precision?

Reply:
When parsing proc on startup, second precision is the smallest unit we get. Seconds should be good enough to uniquely identify a process. As the start time is part of the process identity (tuple of PID + start time), we need to truncate here to seconds as well.